pkgs/applications/networking/browsers/vivaldi/default.nix

```diff
@@ -10,25 +10,29 @@
 }:
 
 let
-  archUrl = name: arch: "https://downloads.vivaldi.com/stable/${name}_${arch}.deb";
-in
-stdenv.mkDerivation rec {
-  version    = "1.1";
-  debversion = "stable_1.1.453.47-1";
-  product    = "vivaldi";
-  name       = "${product}-${version}";
+  isLinux64 = stdenv.system == "x86_64-linux";
```

Comment:
Seeing as you wish to use explicit branching, I think something like this is better
```nix
if stdenv.system == "x86_64-linux"
  then { ...}
else if stdenv.system == "i686-linux"
  then { ...}
else
  throw "unsupported system: ${stdenv.system}"
```